tests/robot-tests/tests/libs/slack.py

```diff
@@ -0,0 +1,96 @@
+import os
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+import shutil
+import json
+import requests
+from bs4 import BeautifulSoup
+
+PATH = f'{os.getcwd()}{os.sep}test-results'
+
+
+def _generate_slack_attachments():
+    failed_tests = dict()
+    passed_tests = dict()
+    skipped_tests = dict()
+
+    report = open(f'{PATH}{os.sep}output.xml', 'rb')
+    contents = report.read()
+
+    soup = BeautifulSoup(contents, 'xml')
+    test = soup.find('total').find('stat')
+
+    failed_tests = test['fail']
+    passed_tests = test['pass']
+    skipped_tests = test['skip']
```

Comment:
Very nice! Looked pretty easy in the end :smile:

Reply:
Yep haha, had a few issues with Beautiful soup initially. It seemed to not like local HTML files and wouldn't read the local `report.html` file so Mark suggested getting the info from XML which turned out to be pretty simple